pkg/inject/inject.go

```diff
@@ -427,14 +451,75 @@ func (conf *ResourceConfig) injectPodSpec(patch *Patch) {
 		}
 	}
 
-	sidecar.Env = append(sidecar.Env, v1.EnvVar{
-		Name:  envIdentityDisabled,
-		Value: identityDisabledMsg,
-	})
-	if idctx := conf.configs.GetGlobal().GetIdentityContext(); idctx != nil {
-		log.Warn("Ignoring Identity configuration.")
+	idctx := conf.configs.GetGlobal().GetIdentityContext()
+	if idctx == nil {
+		sidecar.Env = append(sidecar.Env, v1.EnvVar{
+			Name:  envIdentityDisabled,
+			Value: identityDisabledMsg,
+		})
+		patch.addContainer(&sidecar)
+		return
 	}
 
+	sidecar.Env = append(sidecar.Env, []v1.EnvVar{
+		{
+			Name:  envIdentityDir,
+			Value: k8s.MountPathEndEntity,
+		},
+		{
+			Name:  envIdentityTrustAnchors,
+			Value: idctx.GetTrustAnchorsPem(),
+		},
+		{
+			Name:  envIdentityTokenFile,
+			Value: k8s.IdentityServiceAccountTokenPath,
+		},
+		{
+			Name:  envIdentitySvcAddr,
+			Value: conf.proxyIdentityAddr(),
+		},
+		{
+			Name:      "_pod_sa",
+			ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "spec.serviceAccountName"}},
+		},
+		{
+			Name:  "_l5d_ns",
+			Value: conf.configs.GetGlobal().GetLinkerdNamespace(),
+		},
+		{
+			Name:  "_l5d_trustdomain",
+			Value: idctx.GetTrustDomain(),
+		},
+		{
+			Name:  envIdentityLocalName,
+			Value: "$(_pod_sa).$(_pod_ns).serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)",
+		},
+		{
+			Name:  envIdentitySvcName,
+			Value: "linkerd-identity.$(_l5d_ns).serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)",
+		},
+		{
+			Name:  envDestinationSvcName,
+			Value: "linkerd-controller.$(_l5d_ns).serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)",
+		},
+	}...)
+
+	if len(conf.podSpec.Volumes) == 0 {
+		patch.addVolumeRoot()
```

Comment:
Now that you're using this func, you can remove this comment from `pkg/inject/patch.go`:

```
// TODO use with new identity injection
//nolint:unused
```